qml/main.qml

```diff
@@ -139,6 +139,24 @@ ApplicationWindow {
                 && (settings.desktopAvailableHeight === Screen.desktopAvailableHeight)
         app.x = (savedScreenLayout) ? settings.x : Screen.width / 2 - app.width / 2
         app.y = (savedScreenLayout) ? settings.y : Screen.height / 2 - app.height / 2
+
+        // Check if app.x and app.y are inside any of the monitors
+        for (var i = 0; i < monitorList.length; i+=2)  {
+            var minx = monitorList[i].first
+            var maxx = monitorList[i].second
+            var miny = monitorList[i+1].first
+            var maxy = monitorList[i+1].second
+
+            if (app.x > minx && app.x < maxx) {
+                if (app.y > miny && app.y < maxy) {
+                    return
+                }
+            }
+        }
+
```

Comment:
We should probably add a comment here, that this is the fallback when the saved position seems to be outside of the available screen geometry, and we then just put the app in the middle of the screen.

Reply:
I agree, but I think even better would be to refactor the early return. For example by using a nested function:

```
        function windowPositionIsInsideSomeMonitor() {
            for (var i = 0; i < monitorList.length; i+=2)  {
                var minx = monitorList[i].first
                var maxx = monitorList[i].second
                var miny = monitorList[i+1].first
                var maxy = monitorList[i+1].second

                if (app.x > minx && app.x < maxx) {
                    if (app.y > miny && app.y < maxy) {
                        return
                    }
                }
            }
        }

        if (!windowPositionIsInsideSomeMonitor()) {
            app.x = Screen.width / 2 - app.width / 2
            app.y = Screen.height / 2 - app.height / 2
        }
```